### DISCUSSION TITLE
Cockpit vs Grafana Loki+Promtail

### DISCUSSION THREAD

**Discussion by FAUSTMANNSTEF** [cockpit-project/cockpit#21330] [Q&A]
Hey guys !

I am currently working on a project  aimed to monitor servers from a centralized environment and i came across cockpit .

I can already see that cockpit works amazingly for monitoring a servers performance ( system health checks , error logs, warning logs etc.). However i was wondering if i can use cockpit instead of grafana to adress these issues :

1) Insight into **Nginx logs** ( clear view of both old and new Nginx Access and Error logs + monitor any other additional services running on the server )

2) Monitor **docker internal and service logs** (gain insight into the docker engine + applications running within docker containers.

3) **Freedom** for custom logs for project specific stuff i would like to monitor ( is Cockpit bound to specific logs or can i delve deeper into more detailed/custom stuff if i wanted to? )

4) Can cockpit run alongside each project on the server it is hosted on? in case i would like to monitor multiple projects from a centralized environment ?.

5) Is it **scalable** ? ( if i create a new project which i would like to monitor is it easy to integrate with the existing cockpit setup that contains projects 

6) Ability to set up **trackers** and **slack integration** ? (   We aim to set up trackers for our APIs to check on data accessibility and CVE trackers for the supporting software we use in our projects . Would it also be possible to integrate central logging into slack ?

Thank you very much for your time guys !

I appreciate any source or answer regarding this post **!**

**Comment by jelly** [comment]
Cockpit's metrics and logs are per host and have no reporting / tracking capabilities nor will they because you can only view your logs when you connect to Cockpit there is no background daemon.

In short:

- The metrics are shown from the currently connected host
- The logs all come from journald so if your application logs to journald it is shown
- There is no reporting functionality

**Reply by FAUSTMANNSTEF** [reply]
Thank you very much for your input jelly ! Do you have any experience with also checking older logs using cockpit ? What i mean is in grafana for example i can have a time series database with my custom logs and it would be pretty easy to display them and see whats going on. Is it also this easy when using cockpit ( essentially journald)

**Reply by jelly** [reply]
Have you tried Cockpit, it can display logs from periods X to Y. But that depends all on how long you retain logs, Grafana doesn't use a time series database for logs but Loki which might retain logs longer.
